### PR TITLE
Add host extension httpRequest

### PR DESCRIPTION
**Pull request by kvinwang**
This PR adds one more host API: httpRequest which allows sending HTTP requests in JS.

### CODE REVIEW

**Comment by Leechael** [comment]
1. Can we add a timeout for a request?
2. HEAD / OPTIONS request may not include body.

**Comment by kvinwang** [comment]
> * Can we add a timeout for a request?

In the context of an ink call, all codes must be finished in 10s, so I think a timeout doesn't make much sense in this situation.

> * HEAD / OPTIONS request may not include body.

Yes, all fields are optional except the `url`.

**Comment by Leechael** [comment]
lgtm.

File: README.md [hunk 1]
```diff
@@ -115,6 +117,40 @@ const contractOutput = pink.invokeContract({
...
+```
+
+## Error handling
+Host calls would raise an exception if any error is encountered.
+For example, if we pass an invalid method to the API:
+```js
+try {
+  const response = pink.httpReqeust({
+    url: "https://httpbin.org/ip",
+    method: "GET",
```
Comment (h4x3rotab) [inline README.md]: Why is GET invalid?

**Review by h4x3rotab — APPROVED** [review]
LGTM

File: README.md (hunk 1 above)
Comment (kvinwang) [reply README.md]: A mistake. Fixed.

**Pull request merged**